source/adapters/cuda/image.cpp

```diff
@@ -430,6 +430,11 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageFreeExp(
   ScopedContext Active(hDevice);
   try {
     UR_CHECK_ERROR(cuArrayDestroy((CUarray)hImageMem));
+    if (auto it = hDevice->ChildCuarrayFromMipmapMap.find((CUarray)hImageMem);
+        it != hDevice->ChildCuarrayFromMipmapMap.end()) {
```

Comment:
This seems to be a bug. You are ignoring the result of a check for `ChildCuarrayFromMipmapMap.end()`, as it is placed inside of the conditional.